### ISSUE TITLE
Investigate gpu.js

### ISSUE THREAD

**Issue by samreid** [phetsims/wave-interference#25]
As part of #3 we should investigate https://github.com/gpujs/gpu.js, it advertises itself as a subset of JavaScript that can be compiled to shaders or run in JS, it could be a way for us to write the algorithm once and get it to run on GPU with a CPU fallback.  We should at least check the performance characteristics on iPad2 (I don't know how they get data out of the WebGL, or if they require floating point textures).

**Comment by samreid** [comment]
In testing this out, I ran into https://github.com/gpujs/gpu.js/issues/241.  The main point of gpu.js is to factor out the boilerplate of encoding data and running it on the GPU.  It uses a parser to take a subset of JS and transpile to JS and shader code.  You cannot debug the kernel directly, you would need to write it separately in subset of JS for debugging.

I tried to tinker with understanding the issue mentioned above, but I'm running into a problem where the shader is cached or something (cannot easily change and test shader code).
